src/main/diagnostics/dockerCliSymlinks.ts

```diff
@@ -40,71 +40,85 @@ export class CheckerDockerCLISymlink implements DiagnosticsChecker {
   readonly access = fs.promises.access;
 
   async check() {
-    const rdBinPath = path.join(paths.integration, this.name);
     const dockerCliPluginDir = path.join(os.homedir(), '.docker', 'cli-plugins');
-    let passed = false;
+    const startingPath = path.join(dockerCliPluginDir, this.name);
+    const rdBinPath = path.join(paths.integration, this.name);
+    const finalTarget = path.join(paths.resources, os.platform(), 'bin', this.name);
+    const finalDisplayableTarget = replaceHome(finalTarget);
     let state;
+    let link = '';
+    let description = `The file ${ startingPath }`;
+    let finalDescription = '';
 
     try {
-      let link = await this.readlink(path.join(dockerCliPluginDir, this.name));
-
-      console.debug(`docker-cli symlink: ${ this.name }: first-level symlink: ${ link } (expect ${ rdBinPath })`);
-      if (link === rdBinPath) {
-        while (true) {
-          try {
-            link = await this.readlink(link);
-          } catch (ex) {
-            break;
-          }
-        }
-        console.debug(`docker-cli symlink: ${ this.name }: final symlink: ${ link } (app dir ${ paths.resources })`);
-        if (path.relative(paths.resources, link).startsWith('.')) {
-          state = `is a symlink to ${ replaceHome(link) }, which is not from Rancher Desktop`;
-        } else {
-          try {
-            await this.access(link, fs.constants.X_OK);
-            state = `is a symlink to ~/.rd/${ this.name }`;
-            passed = true;
-          } catch (ex) {
-            const code = (ex as any).code ?? '';
-
-            if (code === 'ENOENT') {
-              state = `is a symlink to ${ replaceHome(link) }, which does not exist`;
-            } else if (code === 'ELOOP') {
-              state = `is a symlink with a loop`;
-            } else if (code === 'EACCES') {
-              state = `is a symlink to ${ replaceHome(link) }, which is not executable`;
-            } else {
-              state = `is a symlink to ${ replaceHome(link) }, but we could not read it (${ code })`;
-            }
-          }
-        }
-      } else {
-        state = `is a symlink to ${ replaceHome(link) }`;
+      link = await this.readlink(startingPath);
+
+      console.debug(`docker-cli symlink: ${ this.name }: first-level symlink ${ startingPath }: points to: ${ link } (expect ${ rdBinPath })`);
+
+      if (link !== rdBinPath) {
+        return {
+          description: `${ description } should be a symlink to ${ replaceHome(rdBinPath) }, but points to ${ link }.`,
```

Comment:
```suggestion
          description: `${ description } should be a symlink to ${ replaceHome(rdBinPath) }, but points to ${ replaceHome(link) }.`,
```

I'm basically thinking of it like raw paths are a string that requires normalizing before emitting to the user, similar to strings that need escaping.

Reply:
yes